mockingbird/src/commonMain/kotlin/com/careem/mockingbird/test/Capture.kt

```diff
@@ -47,18 +49,59 @@ public fun <T> capture(list: CapturedList<T>): CapturedMatcher<T> {
  */
 
 public class Slot<T> : Captureable {
+
+    private val genericSlot : GenereicSlot<T> = initializeSlot<T>()
```

Comment:
I wonder if we can do `: Capturable by genericSlot`

Reply:
I would say the slot is already Capturable.  I think you can do something like : 
`: GenericSlot by genericSlot`, but I would say the property delegation creates overhead (tried with ReadOnlyProperty).

Reply:
Does it create more overhead than manual call forwarding though? I'd be surprised if it does.

Reply:
@sergey135 yep it's alrady capturable, I was suggesting to avoid manual call forwarding by using `by`.